programs/fileio.c

```diff
@@ -91,6 +91,32 @@ static clock_t g_time = 0;
 
 #define MIN(a,b)    ((a) < (b) ? (a) : (b))
 
+#if defined(_MSC_VER) && _MSC_VER >= 1400
+#   define LONG_SEEK _fseeki64
+#elif defined(__MINGW32__) && !defined(__STRICT_ANSI__) && !defined(__NO_MINGW_LFS) && defined(__MSVCRT__)
+#   define LONG_SEEK fseeko64
+#elif defined(_WIN32) && !defined(__DJGPP__)
```

Comment:
Pay attention to maintenance cost. It is important too, even if only maintainers pay it with their time. Any line of code added must be worth it.

Dropping a non-negligible amount of lines to be maintained for the benefit of a rare (dying) configuration which doesn't work anyway due to other reasons, this feels like a bad investment.

Now, it seems your patch also benefits MinGW in `__STRICT_ANSI__` mode, which is a more palatable target. 
For this reason, I'll merge it.